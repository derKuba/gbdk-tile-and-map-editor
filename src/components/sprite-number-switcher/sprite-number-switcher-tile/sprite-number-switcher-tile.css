.sprite-number-switcher-tile {
  width: 40px;
  height: 40px;
  border: 1px solid black;
  display: flex;

}

.sprite-number-switcher-tile  > span {
  align-self: center;
  margin: 0 auto;
}

.active {
  background-color: #33ccff;

}