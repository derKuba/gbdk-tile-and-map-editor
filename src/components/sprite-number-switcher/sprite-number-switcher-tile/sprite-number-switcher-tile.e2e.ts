import { newE2EPage } from '@stencil/core/testing';

describe('mvjp-sprite-number-switcher-tile', () => {
    it('renders', async () => {
        const page = await newE2EPage();
        await page.setContent('<mvjp-sprite-number-switcher-tile></mvjp-sprite-number-switcher-tile>');

        const element = await page.find('mvjp-sprite-number-switcher-tile');
        expect(element).toHaveClass('hydrated');
    });
});
