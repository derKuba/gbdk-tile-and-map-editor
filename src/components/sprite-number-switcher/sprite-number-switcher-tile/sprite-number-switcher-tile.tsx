import { Component, Prop, h } from '@stencil/core';

@Component({
    tag: 'mvjp-sprite-number-switcher-tile',
    styleUrl: 'sprite-number-switcher-tile.css',
    shadow: false,
})
export class SpriteNumberSwitcherTile {
    /**
     *   index of tile in array
     */
    @Prop() index: number;

    /**
     *   active state
     */
    @Prop() isActive: boolean;

    /**
     *  handle click event
     */
    @Prop() clickTile: Function;

    render() {
        const spriteIsActive = `${this.isActive ? 'active' : ''}`;
        return (
            <div class={`sprite-number-switcher-tile ${spriteIsActive}`} onClick={() => this.clickTile(this.index)}>
                <span>{this.index}</span>
            </div>
        );
    }
}
