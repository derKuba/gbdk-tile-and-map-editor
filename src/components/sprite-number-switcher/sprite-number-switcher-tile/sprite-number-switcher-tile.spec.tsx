import { newSpecPage } from '@stencil/core/dist/testing';
import { h } from '@stencil/core';

import { SpriteNumberSwitcherTile } from './sprite-number-switcher-tile';

describe('mvjp-sprite-number-switcher-tile', () => {
    it('builds', () => {
        expect(new SpriteNumberSwitcherTile()).toBeTruthy();
    });

    it('matches snapshot', async () => {
        const page = await newSpecPage({
            components: [SpriteNumberSwitcherTile],
            template: () => {
                return (
                    <div>
                        <mvjp-sprite-number-switcher-tile index={0} isActive={false} clickTile={() => {}} />
                        <mvjp-sprite-number-switcher-tile index={1} isActive={true} clickTile={() => {}} />
                        <mvjp-sprite-number-switcher-tile index={2} isActive={false} clickTile={() => {}} />
                    </div>
                );
            },
        });

        expect(page.root).toMatchSnapshot();
    });

    it('matches snapshot', async () => {
        const clickTile = jest.fn();
        const page = await newSpecPage({
            components: [SpriteNumberSwitcherTile],
            template: () => {
                return (
                    <div>
                        <mvjp-sprite-number-switcher-tile index={23} isActive={false} clickTile={clickTile} />
                        <mvjp-sprite-number-switcher-tile index={1} isActive={true} clickTile={() => {}} />
                        <mvjp-sprite-number-switcher-tile index={2} isActive={false} clickTile={() => {}} />
                    </div>
                );
            },
        });
        console.log(page.root.innerHTML);
        page.root.querySelector('.sprite-number-switcher-tile').click();

        expect(clickTile).toBeCalledWith(23);
    });
});
