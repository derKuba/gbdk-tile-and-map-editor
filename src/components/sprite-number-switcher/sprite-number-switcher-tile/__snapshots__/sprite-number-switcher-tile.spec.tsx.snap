// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mvjp-sprite-number-switcher-tile matches snapshot 1`] = `
<mvjp-sprite-number-switcher-tile>
  <div class="sprite-number-switcher-tile">
    <span>
      0
    </span>
  </div>
</mvjp-sprite-number-switcher-tile>
`;
